### PR TITLE
511-feature-request-nikto-tool-pull-request

### PR DESCRIPTION
**Pull request by SethJ1**
Had to redo a pull request due to certain unwanted files being added, old commits are as follows: 
image

### CODE REVIEW

File: src/components/Nikto/Nikto.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import { useState } from "react";
+import { Button, Checkbox, LoadingOverlay, Stack, TextInput } from "@mantine/core";
+import { useForm } from "@mantine/form";
+import { CommandHelper } from "../../utils/CommandHelper";
+import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
+import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { UserGuide } from "../UserGuide/UserGuide";
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:7]: - [ ] Upgrade to render component

File: src/components/Nikto/Nikto.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+// Define title and user guide description for the Nikto tool
+const title = "Nikto";
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:10]: - [ ] Move inside function definition.

File: src/components/Nikto/Nikto.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+const description_userguide =
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:11]: - [ ] Seperate into guide and description variables required for render component

File: src/components/Nikto/Nikto.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    "Nikto is a powerful web server scanner that can perform comprehensive tests against web servers for multiple items.\n" +
+    "How to use this Nikto tool:\n" +
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:13]: - [ ] Use steps method (Step 1: ...)

File: src/components/Nikto/Nikto.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    "- Provide the target URL or IP address to scan.\n" +
+    "- Start the scan to gather information about potential vulnerabilities and misconfigurations.\n" +
+    "- Review the scan output to identify any security issues.\n";
+
+// Define the type for form values
+interface FormValuesType {
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:19]: - [ ] Doc String required

File: src/components/Nikto/Nikto.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    TargetURL: string;
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:20]: - [ ] camelCase

File: src/components/Nikto/Nikto.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    sslScan: boolean;
+}
+
+// Define the Nikto component
+const Nikto = () => {
+    // Initialize state variables
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:26]: - [ ] Check comment consistency with other components.
// Component State Variables.

File: src/components/Nikto/Nikto.tsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    const [loading, setLoading] = useState(false);
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:27]: - [ ] Inline comment

File: src/components/Nikto/Nikto.tsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    const [output, setOutput] = useState("");
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:28]: - [ ] Inline comment

File: src/components/Nikto/Nikto.tsx [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    const [allowSave, setAllowSave] = useState(false);
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:29]: - [ ] Inline comment

File: src/components/Nikto/Nikto.tsx [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    const [hasSaved, setHasSaved] = useState(false);
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:30]: - [ ] Inline comment

File: src/components/Nikto/Nikto.tsx [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+    const [sslScan, setSslScan] = useState(false);
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:31]: - [ ] Inline comment

File: src/components/Nikto/Nikto.tsx [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+    // Initialize form state using the useForm hook
+    let form = useForm<FormValuesType>({
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:34]: - [ ] Change to a constant.
- [ ] Create an instance of not specify.

File: src/components/Nikto/Nikto.tsx [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+        initialValues: {
+            TargetURL: "",
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:36]: - [ ] camelCase

File: src/components/Nikto/Nikto.tsx [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+            sslScan: false,
+        },
+    });
+
+    // Function to handle form submission
+    const onSubmit = async (values: FormValuesType) => {
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:42]: - [ ] Doc String

File: src/components/Nikto/Nikto.tsx [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+        } catch (error: any) {
+            setOutput(`Error: ${error.message}`);
+        } finally {
+            // Set loading state to false and allow output saving
+            setLoading(false);
+            setAllowSave(true);
+        }
+    };
+
+    // Function to handle completion of output saving
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:66]: - [ ] Doc string

File: src/components/Nikto/Nikto.tsx [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+    const handleSaveComplete = () => {
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:67]: - [ ] doc string

File: src/components/Nikto/Nikto.tsx [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+    // Function to clear command output
+    const clearOutput = () => {
+        setOutput("");
+        setHasSaved(false);
+        setAllowSave(false);
+    };
+
+    // Render component
+    return (
+        <form onSubmit={form.onSubmit(onSubmit)}>
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:81]: - [ ] RenderComponent missing
- [ ] Dependency runtime missing

File: src/components/Nikto/Nikto.tsx [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+            <Stack>
+                {UserGuide(title, description_userguide)}
+                <TextInput label="Target URL" required {...form.getInputProps("TargetURL")} />
+                <Checkbox
+                    label="SSL Scan"
+                    checked={sslScan}
+                    onChange={(event) => setSslScan(event.currentTarget.checked)}
+                />
+                <Button type="submit" disabled={loading}>
+                    Start Nikto Scan
```
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:92]: - [ ] <Button type={"submit"}>Start {title}</Button>

File: src/components/Nikto/Nikto.tsx
Comment (bradleybogg) [inline src/components/Nikto/Nikto.tsx:1]: Look at a already baselined component for reference such as AirBase NG

**Review by bradleybogg — CHANGES_REQUESTED** [review]
Good Work @SethJ1 

Please see the below changes.

Kind Regards

**Review by bradleybogg — CHANGES_REQUESTED** [review]
- [x] Install dependency checker
- [ ] Fix merge conflicts

**Comment by michaeljpigott** [comment]
@zkkok This one is ready for application testing.

**Comment by zkkok** [comment]
Hi @SethJ1,
Please resolve the conflicts before application testing

**Comment by SethJ1** [comment]
@zkkok all done

File: src/components/Nikto/Nikto.tsx [hunk 20]
```diff
@@ -0,0 +1,192 @@
...
+        setLoading(true);
+
+        // Construct arguments for the Nikto command based on form input
+        const args = ["-h", values.targetURL];
+        if (values.sslScan) {
+            args.push("-ssl"); // Add -ssl option for SSL scanning
+        }
+
+        // Execute the Nikto command via helper method and handle its output or potential errors
+        CommandHelper.runCommandWithPkexec("nikto", args, handleProcessData, handleProcessTermination)
```
Comment (zkkok) [inline src/components/Nikto/Nikto.tsx:127]: Replace runCommandWithPkexec with runCommandGetPidAndOutput

**Review by zkkok — CHANGES_REQUESTED** [review]
@SethJ1 overall great work with the tool, basic functionality test passes. Just a slight change needed before it can be approved. The tool does not need root privileges, please modify it so that it does not use root commands.

**Comment by SethJ1** [comment]
@zkkok done

**Review by zkkok — CHANGES_REQUESTED** [review]
@SethJ1, functionality test looks good, tool worked as expected but the overlay does not show up while running the tool, please review your code again.

**Comment by SethJ1** [comment]
@zkkok I have fixed the overlay.

**Review by zkkok — APPROVED** [review]
The overlay works now

**Pull request merged**